src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -61,20 +106,50 @@ const collectorSignalsLoader = async (
 
   // Handle signals for auction artworks
   if (isInSale && auctionsCollectorSignalsEnabled) {
-    const activeSaleArtwork = await getActiveSaleArtwork(
+    const activeLotData = await getActiveSaleArtwork(
       {
         artworkId,
         saleIds: artwork.sale_ids,
       },
       ctx
     )
 
-    if (activeSaleArtwork) {
+    const activeAuctionLot = activeLotData && !activeLotData.sale.ended_at
+    if (activeAuctionLot) {
+      const { saleArtwork, sale } = activeLotData
+
       if (artwork.recent_saves_count) {
         lotWatcherCount = artwork.recent_saves_count
       }
-      if (activeSaleArtwork.bidder_positions_count) {
-        bidCount = activeSaleArtwork.bidder_positions_count
+
+      if (saleArtwork.bidder_positions_count) {
+        bidCount = saleArtwork.bidder_positions_count
+      }
+
+      const futureRegistrationEndTime = ifFutureDate(sale.registration_ends_at)
+      if (futureRegistrationEndTime) {
```

Comment:
#non-blocking: `ifFutureDate` returns a date or `null`, and I guess you can just do assignment directly? Or I wonder if it's more clear to have it return a boolean like `isFuture`. (Not sure what's our current recommendation about date/time libraries, but there could already be helper for this.)

Reply:
My hope here was to have some filtering/conditional logic rolled into one - if you pass a date + it is in the future, you know you can work with it from there forwards without needing to say `if (something?.some_date && isFuture(something.some_date)) { const futureDate = something.some_date ; /* ... */ }`.

Updated this function to be a little more explicit about what it does.